pkg/agent/apiserver/apiserver.go

```diff
@@ -74,6 +74,15 @@ func (s *agentAPIServer) Run(stopCh <-chan struct{}) error {
 	return s.GenericAPIServer.PrepareRun().Run(stopCh)
 }
 
+func (s *agentAPIServer) GetCertData() []byte {
+	secureServingInfo := s.GenericAPIServer.SecureServingInfo
+	if secureServingInfo == nil {
+		return nil
+	}
+	cert, _ := secureServingInfo.Cert.CurrentCertKeyContent()
```

Comment:
While I think something like this would work:
```go
block, _ := pem.Decode(cert)
certs := x509.ParseCertificates(block)
intermediates := certs[1:] // distribute this only
```
I am not sure this is the best approach. It feels like it's more common to distribute the cert "file" as is, in the self-signed case. That's also what we do for the Antrea Controller (when we call `dynamiccertificates.NewDynamicCAContentFromFile` after generating the certificate). That may be best practice for a standalone self-signed certificate (as opposed to the case where the same self-signed CA is used to generate multiple certificates). I feel more comfortable keeping it this way for now. What do you think?

By renaming the field to `APICABundle` as you suggested, we are free to change the contents of the published data later and remove the leaf certificate.

Reply:
works for me